tracer/src/Datadog.Trace/Iast/EvidenceConverter.cs

```diff
@@ -6,7 +6,10 @@
 #nullable enable
 
 using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.Iast.Helpers;
```

Comment:
Should this documentation be updated with the new pattern field?
```
 // When redacted output is:
    // "valueParts": [
    //   { "value": "SELECT * FROM Users WHERE " },     -> Not tainted part
    //   { "source": 0, "value": "Name='" },            -> Tainted part from source 0
    //   { "source": 0, "redacted": true },             -> Redacted tainted part from source 0
    //   { "source": 0, "value": "'" },                 -> Tainted part from source 0
    //   { "value": " and RoleId=" },                   -> Not tainted part
    //   { "redacted": true }                           -> Redacted not tainted part
    // ]
```

Also this class contains a lot of algorithm and is quite hard to grasp, so comments are welcome :)

Reply:
Agree